### ISSUE TITLE
Add Cloudflare Turnstile support for better user experience.

### ISSUE THREAD

**Issue by nativehyun** [mastodon/mastodon#26468]
### Pitch

Greetings!
I saw #25019 for optional hCaptcha Support and I think it's really great idea for blocking spam from bots.
But there is a better solution for user experience - [Cloudflare Turnstile](https://www.cloudflare.com/products/turnstile/).

### Motivation

With hCaptcha, users are need to solve difficult captcha problem to proof I'm human.
This is good deal for blocking bots, but users can feel annoying experience. With advances in AI, CAPTCHA may become more difficult in the future.
**There is a better solution to block bots but user don't need to proof themself.** Cloudflare Turnstile.
With Turnstile, sign up can be much faster because Turnstile don't require user to proof. 
But instance admin can still block spam bots.

I think this feature will improve the user experience.
**And starting hCaptcha, I hope there will be more ways to block spam in mastodon.**

Thanks for reading. Feel free to comment and have a great day!

**Comment by iam-py-test** [comment]
I agree CAPTCHAs are annoying, bad UX, and hinder accessability, and hCaptcha specifically has some privacy concerns<sup>[[1]](https://github.com/mastodon/mastodon/issues/25025)</sup>. However, I do not think CloudFlare is the solution, as they are against user privacy and freedom<sup>[[2]](https://crimeflare.eu.org/).

**Comment by nativehyun** [comment]
> I agree CAPTCHAs are annoying, bad UX, and hinder accessability, and hCaptcha specifically has some privacy concerns[[1]](https://github.com/mastodon/mastodon/issues/25025). However, I do not think CloudFlare is the solution, as they are against user privacy and freedom[[2]](https://crimeflare.eu.org/). Thank you.

Thanks for your kind reply!
Every CAPTCHA solution may cause privacy concerns. I think that level of privacy should be determined by the instance administrator. If you don't like hCaptcha or Cloudflare Turnstile, you can go for another instance that values privacy instead of one that values user experience. Because that is the strength of Fediverse.
